packages/dtslint/src/checks.ts

```diff
@@ -130,3 +132,39 @@ export function checkTsconfig(dirPath: string, options: CompilerOptionsRaw): str
   }
   return errors;
 }
+
+export interface AttwResult {
+  status: "pass" | "fail" | "error";
+  output: string;
+}
+
+export function runAreTheTypesWrong(dirPath: string, implementationTarballPath: string, configPath: string): AttwResult {
+  const packageJsonContent = readJson(joinPaths(dirPath, "package.json"));
+  const mangledName = packageJsonContent.name.replace(/^@types\//, "");
+  const tarballName = `types-${mangledName}-${packageJsonContent.version}.tgz`;
+  const attwPackageJsonPath = require.resolve("@arethetypeswrong/cli/package.json");
+  const attwBinPath = joinPaths(dirname(attwPackageJsonPath), readJson(attwPackageJsonPath).bin.attw);
+  const npmPath = which.sync("pnpm", { nothrow: true }) || which.sync("npm");
+  execSync(`${npmPath} pack`, { cwd: dirPath, stdio: "ignore", env: { ...process.env, COREPACK_ENABLE_STRICT: "0" } });
```

Comment:
Does this function have to be synchronous, or can we use the exec helper from `utils`?

Reply:
I guess it could be async, but this should be the only thing happening in the process, so is there an advantage?

Reply:
Using the shared helper would be nice, but that helper is a little wonky so I don't really mind.

I think mainly I don't like using the `child_process` APIs which go through the shell, so using `execFileSync` would be my preferred choice for any execs in the repo.

Reply:
I think I used `execSync` intentionally before I realized `which` was going to be required. I can change it, but it’s kind of annoying because the way of processing output is slightly different for every API

Reply:
All in all I wish we would wholesale switch to `execa`, which has a really, really good API and cross-OS behavior.